website/src/controllers/groups.js

```diff
@@ -761,6 +761,12 @@ api.removeMember = function(req, res, next){
         update.$unset['newMessages.' + group._id] = '';
         User.update({_id: removedUser._id, apiToken: removedUser.apiToken}, update).exec();
 
+        var update = {$pushAll:{
+          "flags.kickedFromGroup":[{"name":group.name, "message": message}]
+        }};
+
+        User.update({_id: removedUser._id, apiToken: removedUser.apiToken}, update).exec();
```

Comment:
Also, might be better to use: http://mongoosejs.com/docs/api.html#query_Query-findOneAndUpdate